tests/tests_api/test_api.py

```diff
@@ -102,3 +68,79 @@ def test_throw_error_at_source_registration_if_error(
         kirby_app.add_destination(MagicMock(id=1))
     with pytest.raises(ClientError):
         kirby_app.add_destination(MagicMock(id=-1))
+
+
+@patch("requests.session")
+def test_throw_error_at_source_registration_if_error(
+    session_mock, kirby_expected_env, kirby_hidden_env
+):
+    session_mock.return_value.get.side_effect = [
```

Comment:
instead of providing a list, I'd use the pytest.mark.parametrize and provide different MagicMock values instead. So that you can extend your test from the outside, and the actual test codes becomes slimmer

Reply:
The test is now rewritten